### PR TITLE
Move tab logic outside of editor panel

### PR DESCRIPTION
**Pull request by yann300**

### CODE REVIEW

File: src/app/files/fileManager.js [hunk 1]
```diff
@@ -101,6 +99,18 @@ class FileManager {
     return self._deps.config.get('currentFile')
   }
 
+  closeFile (name) {
+    var self = this
```
Comment (GrandSchtroumpf) [inline src/app/files/fileManager.js]: I think you can use `this` with confidence here.

File: src/app/files/fileManager.js [hunk 2]
```diff
@@ -101,6 +99,18 @@ class FileManager {
...
+    delete self.openedFiles[name]
+    if (Object.keys(self.openedFiles).length) {
+      self.switchFile(Object.keys(self.openedFiles)[0])
+    } else {
+      self._deps.editor.displayEmptyReadOnlySession()
+      self._deps.config.set('currentFile', '')
+    }
+    this.event.trigger('fileClosed', [name])
```
Comment (GrandSchtroumpf) [inline src/app/files/fileManager.js]: Should you use `this.nodeEvent.emit` instead ?

File: src/app/panels/tab-proxy.js [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+      }
+    }
+  }
+
+  addHandler (type, fn) {
+    this.handlers[type] = fn
+  }
+
+  renderTabsbar () {
+    var self = this
```
Comment (GrandSchtroumpf) [inline src/app/panels/tab-proxy.js]: You can use `this`

File: src/app/panels/tab-proxy.js [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  constructor (fileManager, editor) {
+    this.fileManager = fileManager
+    this.editor = editor
+    this.activeEntity
+    this.entities = {}
+    this.data = {}
+    this._view = {}
+
+    fileManager.event.register('fileRemoved', (name) => {
+      var filesEl = document.querySelector('#files')
```
Comment (GrandSchtroumpf) [inline src/app/panels/tab-proxy.js]: You can use `const`

File: src/app/panels/tab-proxy.js [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+export class TabProxy {
+  constructor (fileManager, editor) {
+    this.fileManager = fileManager
+    this.editor = editor
+    this.activeEntity
+    this.entities = {}
+    this.data = {}
+    this._view = {}
+
+    fileManager.event.register('fileRemoved', (name) => {
```
Comment (GrandSchtroumpf) [inline src/app/panels/tab-proxy.js]: If used with `nodeEvent` use `on` instead of `register`

File: src/app/files/fileManager.js (hunk 2 above)
Comment (yann300) [reply src/app/files/fileManager.js]: yeah, fileManager is already using the remix `EventManager` for the `currentFileChanged` event. better to keep using that one for now and not mixing event...

**Pull request merged**